### PR TITLE
Add missing mappings for NVRTC and cuSparse

### PR DESCRIPTION
**Pull request by lcskrishna**
This PR contains the APIs that are missing in cuda_to_hip_mappings for NVRTC and cuSparse libraries that are required for CuPy development.

### CODE REVIEW

File: hipify_torch/cuda_to_hip_mappings.py
Comment (AdrianAbeyta) [inline hipify_torch/cuda_to_hip_mappings.py:7896]: Please check into the difference between CONV_MATH_FUNC and CONV_TYPE it seems like some _t API's are labeled as  CONV_MATH_FUNC.

**Review by AdrianAbeyta — CHANGES_REQUESTED** [review]
Please check that existing and current entries follow the same syntax. I.E all types are labeled as "CONV_TYPE" and not "CONV_MATH_FUNC". Apart from that it LGTM.

File: hipify_torch/cuda_to_hip_mappings.py
Comment (lcskrishna) [reply hipify_torch/cuda_to_hip_mappings.py]: Done.

**Review by AdrianAbeyta — APPROVED** [review]
LGTM, please wait for hipify_torch folks to approve before merging.

**Review by jithunnair-amd — APPROVED** [review]
LGTM

**Comment by jithunnair-amd** [comment]
@lcskrishna Merging this, but for the record, please mention here which branch of Pytorch you tested these changes with, at least for the Pyatorch build step.

**Comment by lcskrishna** [comment]
@jithunnair-amd These mappings are needed for CuPy. So tested against the CuPy codebase.

**Pull request merged**
